core/execution/engine.go

```diff
@@ -683,18 +851,45 @@ func (e *Engine) propagateInitialNetParamsToFutureMarket(ctx context.Context, mk
 
 func (e *Engine) removeMarket(mktID string) {
 	e.log.Debug("removing market", logging.String("id", mktID))
-
-	delete(e.markets, mktID)
-	for i, mkt := range e.marketsCpy {
+	delete(e.commonMarkets, mktID)
+	for i, mkt := range e.commonMarketsCpy {
+		m := e.commonMarketsCpy[i]
+		println(m.GetID())
```

Comment:
Probably want to remove the `println`

Reply:
I feel like a doctor who forgot the scissors in the stomach of the patient :)